### DISCUSSION TITLE
Backtracking options for “split boost”, buffer

### DISCUSSION THREAD

**Discussion by williamhobbs** [williamhobbs/2024_pvpmc_self_shade#10] [Ideas]
I was thinking about how to model “split boost” backtracking for twin modules (I think that’s NexTracker’s name for it) where the bottom half of modules are allowed to be shaded in 1P trackers. 

Setting up so that there’s a second parameter for backtracking that is the shaded fraction limit makes sense to me. In 1P twin modules, split boost tracking would have a limit of fs = 0.5. For 2P, you might set it to 0.25, 0.5, or even 0.75. For conventional backtracking, the limit would be fs = 0. 

This then got me thinking about real world systems, where there’s some sort of “buffer” often added in backtracking to account for imperfections/non-uniformities. That can be done by “padding” a GCR, but it could also be done by setting a small negative fs limit, for example, -0.05 would target 5% of module length as buffer. 

Does this make sense? Worth trying to implement somewhere here? Should I post this on the pvlib GitHub discussions?

**Comment by mikofski** [comment]
@madisonghiz and i have been working on this and we’ve been calling it “half-tracking” we can share what we have and merge it that would be awesome. One thing we hypothesized was that half-tracking doesn’t work as well if the light is too diffuse

**Reply by williamhobbs** [reply]
Cool! 

I did a quick test and it looks like our current code can do this by passing a different GCR to `pvlib.tracking.singleaxis()`. For example, passing GCR*.5 does half tracking. 

The diffuse fraction part makes sense - I had not thought of that (or maybe you mentioned it on a call and I forgot). 

I kind of like the idea of calling it “half-backtracking” for allowing half of the array to be shaded. With 2P twin modules, you could also have “quarter-backtracking”(allowing 3/4 of the array to be shaded) and “three-quarter-backtracking” (allowing only 1/4 of the array to be shaded).

**Reply by kandersolar** [reply]
We called the general concept "fractional backtracking" in https://doi.org/10.1063/5.0202220, and Eq 15 does the (1-fs) * gcr thing.

**Reply by williamhobbs** [reply]
@kandersolar maybe I subconsciously stole your nomenclature 🙂

**Comment by mikofski** [comment]
>For example, passing GCR*.5 does half tracking.

Yes this is exactly what we did. TBH I hadn’t thought of 2P trackers with twin modules, would be interesting to experiment with variations, but I think main focus is 1P.

For diffuse > 0.4 annually, I’ve seen that more limited tracking, say between -25° to 25° or some optimized range is better than true tracking even for thin film, but it’s not a hard & fast rule, depends on *when* it’s cloudy, not only *how much* clouds

**Comment by kandersolar** [comment]
For buffer -- I've thought about this, but I'm not sure that negative target fs gets us anything we didn't have before.  As you noted for split-boost backtracking, it's mathematically equivalent to tweaking the GCR, so it's essentially the usual "adjust the GCR" strategy.  

If the uncertainty that needs accommodation is in the pile height information (seems like this is the main one?), I think it would be better to account for the buffer in the cross-axis slope parameter, at least theoretically...
